cocos/2d/CCMotionStreak.cpp

```diff
@@ -126,7 +127,7 @@ bool MotionStreak::initWithFade(float fade, float minSeg, float stroke, const Co
     _blendFunc = BlendFunc::ALPHA_NON_PREMULTIPLIED;
 
     // shader state
-    setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_COLOR));
+    setGLProgramState(GLProgramState::getOrCreateWithGLProgramName(GLProgram::SHADER_NAME_POSITION_TEXTURE_A8_COLOR));
```

Comment:
You are right, but what i tried to say was that it is an easy fix.
The SHADER_NAME_POSITION_TEXTURE_A8_COLOR shader uses v_fragColor.rgb, if you modify that to use the texture2D rgb values it *should* be fixed